File: src/main/java/com/vc/medium/L921.scala
package com.vc.medium

object L921 {
  def minAddToMakeValid(str: String): Int = {
    import scala.collection.mutable
    val st = new mutable.Stack[Char]()

    str.foreach({
      case '(' => st.push('(')
      case  ')' => {
        if(st.nonEmpty && st.head == '(') st.pop
        else st.push(')')
      }
    })
    st.size
  }
}

File: src/main/java/com/vc/medium/L921.java
package com.vc.medium;

import java.util.*;

class L921 {
    public int minAddToMakeValid(String S) {
        Stack<Character> st = new Stack<>();
        int res = 0;
        for(int i = 0; i < S.length(); i++) {
            char ch = S.charAt(i);
            if(ch == '(') st.push(ch);
            else {
                if(!st.isEmpty()) st.pop();
                else res++;
            }
        }
        return res + st.size();
    }
}
